### PR TITLE
[WIP] add models for SPM event and parametric design tutos

### PR DESCRIPTION
**Pull request by Remi-Gau**
based on and should be merged after #13 

## TODO

- [ ] Make it work with fitlins
- [ ] Finish parametric analysis model

### CODE REVIEW

File: spm_tutorials/spm-FaceRep_smdl.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    {
+                    "Name": "AND",
+                    "Input": [
+                        "face_type==famous",
+                        "repetition_type==1"
+                    ],
+                    "Output": [
+                        "famous_1"
+                    ]
+                    },
```
Comment (Remi-Gau) [inline spm_tutorials/spm-FaceRep_smdl.json:24]: this instruction and the following ones aims to create 4 variables from the 2X2 design of the dataset, but I have no idea if this is the correct way to do this

**Comment by Remi-Gau** [comment]
Currently failing this way:

```bash
$ bash facerep_run_fitlins.sh 

Traceback (most recent call last):
  File "/home/remi/github/BIDS-model-zoo/spm_tutorials/env/bin/fitlins", line 8, in <module>
    sys.exit(main())
  File "/home/remi/github/BIDS-model-zoo/spm_tutorials/env/lib/python3.8/site-packages/fitlins/cli/run.py", line 432, in main
    sys.exit(run_fitlins(sys.argv[1:]))
  File "/home/remi/github/BIDS-model-zoo/spm_tutorials/env/lib/python3.8/site-packages/fitlins/cli/run.py", line 383, in run_fitlins
    graph = BIDSStatsModelsGraph(layout, model_dict)
  File "/home/remi/github/BIDS-model-zoo/spm_tutorials/env/lib/python3.8/site-packages/bids/modeling/statsmodels.py", line 95, in __init__
    self.model = self._load_model(model)
  File "/home/remi/github/BIDS-model-zoo/spm_tutorials/env/lib/python3.8/site-packages/bids/modeling/statsmodels.py", line 118, in _load_model
    validate_model(model)
  File "/home/remi/github/BIDS-model-zoo/spm_tutorials/env/lib/python3.8/site-packages/bids/modeling/statsmodels.py", line 34, in validate_model
    names = Counter([n['name'] for n in model['nodes']])
  File "/home/remi/github/BIDS-model-zoo/spm_tutorials/env/lib/python3.8/site-packages/bids/modeling/statsmodels.py", line 34, in <listcomp>
    names = Counter([n['name'] for n in model['nodes']])
KeyError: 'name'
```

File: spm_tutorials/spm-FaceRep_smdl.json (hunk 1 above)
Comment (Remi-Gau) [reply spm_tutorials/spm-FaceRep_smdl.json]: I may probably be need help with figuring out which transformers to use to do this.

@effigies

**Comment by Remi-Gau** [comment]
Apparently pybids `convolve` has the possibility to support SPM style temporal and dispersion derivatives of the HRF.

Will implement this but I have no idea how / if fitlins knows how to handle those at the contrast stage.

**Comment by jdkent** [comment]
fitlins (e.g., pybids) currently does not handle derivatives. see bids-standard/pybids#837.

**Comment by Remi-Gau** [comment]
Sure let me resolve the conflict and have another look. Then I will let you click the big green button.

**Pull request merged**